### DISCUSSION TITLE
ESP32C3 & WS2812 : no worky

### DISCUSSION THREAD

**Discussion by trosenda** [arendst/Tasmota#20116] [Q&A]
Hi,
wanted to swap out an ESP8266 with one of these :
https://templates.blakadder.com/seeedstudio_XIAO_ESP32C3.html

Took the standard tasmota32c3cdc.bin
Connected the stripe to GPIO 10, configured module for GPIO 10
Tried schemes, simple on/off.....But LEDs stay dark. 
Any advise appreciated.
Thanks

**Comment by barbudor** [comment]
How is your strip connected ?
Normally you can't directly connect a WS2812 strip (which are 5V) to an ESP. See https://tasmota.github.io/docs/WS2812B-and-WS2813/
It may have been working -on the edge- with ESP8266 but you may now be on the other side of the edge and that could explain why it doesn't work.

**Reply by sfromis** [reply]
I've not seen any trouble with WS2812 on ESP32-C3, but as barbudor suggested, working or not when voltages are out of spec may be a question of luck. And of course also exactly how you wired it, and what power is available.

**Reply by trosenda** [reply]
Wired directly. no level shifters. working on different 8266 / 8285 as expected. now searching my bench for levelshifters.

**Reply by sfromis** [reply]
Beware that some level shifters might be too slow for 400 KHz, if designed for low frequency stuff.

One thing to try could be to power the strip with 3.3V instead of 5V (if you have sufficient current available). While this is not great for brightness, it would bring 3.3V signals within range, as the usual spec is that signals should be at least 70% of Vcc, meaning 3.5 V for 5V power.

Or if you can, lower input voltage from 5V to 4.5V, bringing the minimum signal voltage down to 3.15 V.

**Comment by trosenda** [comment]
Must be something else. Tried different GPIOs, but there is no output at all when I select WS2812. With PWM I get an output (tried PWM to doublecheck there is nothing wrong with the setup)
![Screenshot from 2023-12-02 17-38-59](https://github.com/arendst/Tasmota/assets/45146808/57d9cebe-7404-438c-96b4-e065238079e7)

**Reply by Jason2866** [reply]
Something else with your hardware. Neopixel works with C3.

**Reply by sfromis** [reply]
Yeah, just now I'm seeing a short neopixel strip work fine with another ESP32-C3 board. Assuming that the message of your 2nd screen photo above is that nothing happens on the pin you're measuring, I'd recommend a careful re-check of wiring, connections and right pin used with correct config.

**Reply by trosenda** [reply]
Progress. I was testing with "Scheme 5" which is supposed to be the Clock mode, and works on the 8266, I decided to use Scheme 6 (candle) in the hope to see some more/better/different signal on the DMM. While it is a candle on the 8266, Scheme 6 on the esp32 turns out to be the clock mode. Now searching for candle in the documentation also gives a strange result. (same for rainbow, christmas,...) they are all of by one.
![Screenshot from 2023-12-03 10-39-49](https://github.com/arendst/Tasmota/assets/45146808/4db6046a-87d1-43fe-9427-3231dd6625e9)

**Reply by sfromis** [reply]
As a basic test, I'd have recommended Scheme 0, and a static color. But yeah, scheme numbers being changed does sound like a bug.
